### PR TITLE
Site migration: show progress during a migration

### PR DESCRIPTION
**Pull request by andfinally**
Note: This change is behind a feature flag and is part of an in-progress larger project. For context, see pbkcP4-8-p2.

#### Changes proposed in this Pull Request

* Adds a progress view to the migrate section, with a simple item list and progress bar.

image

#### Testing instructions

* Check out this pull request and run Calypso locally, or view it on calypso.live with the flag tools/migrate.
* This PR requires D36080-code and a sandboxed API. See the testing instructions for running D36080-code.
* Viewing calypso.localhost:3000/migrate/, you should see a SiteSelector component allowing you to choose a site to migrate a Jetpack site into.
* When you select a site, you should see another SiteSelector component, listing all the Jetpack sites your account has access to.
* Selecting one of those sites should take you to a confirmation screen.
* Accept the confirmation. You should see a progress screen detailing the steps of the migration.
* When the migration is done you should see a message telling you it is complete.

### CODE REVIEW

File: client/my-sites/migrate/section-migrate.jsx [hunk 1]
```diff
@@ -5,23 +5,25 @@ import React, { Component } from 'react';
...
 import FormattedHeader from 'components/formatted-header';
 import Gridicon from 'components/gridicon';
 import HeaderCake from 'components/header-cake';
 import Main from 'components/main';
+import ProgressBar from 'components/progress-bar';
 import SidebarNavigation from 'my-sites/sidebar-navigation';
 import Site from 'blocks/site';
 import SiteSelector from 'components/site-selector';
-import { Interval, EVERY_MINUTE } from 'lib/interval';
+import { Interval, EVERY_THIRTY_SECONDS } from 'lib/interval';
```
Comment (jonathansadowski) [inline client/my-sites/migrate/section-migrate.jsx]: I don't know why every 30 seconds still feels slow to me. I'd suggest `EVERY_FIFTEEN_SECONDS` or maybe `EVERY_TEN_SECONDS` (neither constant currently exists).

**Comment by andfinally** [comment]
Thanks @jonathansadowski, that's a really good point about the progress list. I think the green checkmark is part of the problem because it signifies completion. I've added a "wrapping up" item for now. I've not seen the spinner used as a bullet elsewhere, and it would also be complicated to implement without creating a spinner gif: the Calypso spinner is a CSS animated div component, but here I'm swapping the `list-image` to show the different states.

image

As mentioned @danhauk I've not implemented the vertical centering here – I wasn't sure if it looked OK on my big screen – but I can add it later if necessary. I'll to figure out how to render the migation start time above the progress bar in a separate PR.

File: client/my-sites/migrate/section-migrate.jsx (hunk 1 above)
Comment (andfinally) [reply client/my-sites/migrate/section-migrate.jsx]: Thanks, I've reduced the interval to 10.

**Comment by danhauk** [comment]
I agree about the green checkmark on an in progress task is unclear. I like @jonathansadowski suggestion about using the spinner to indicate the task in progress.

As far as implementation goes, instead of using classes can we switch out the component based on state [here](https://github.com/Automattic/wp-calypso/pull/38053/files#diff-5a90bc5b94ad276cd1f1d73804034db2R233)?

* if it's in progress, render `<Spinner/>`
* if complete, render `<Gridicon icon="checkmark"/>`

```
let backupIcon = <Spinner />;
let restoreIcon;

if ( 'restoring' === this.state.migrationStatus ) {
    backupIcon = <Gridicon icon="checkmark-circle" />;
    restoreIcon = <Spinner />;
}

...

<li className={ backingUpClass }>{ backupIcon } Backing up from { sourceSiteDomain }</li>
<li className={ restoringClass }>{ restoreIcon } Restoring to { targetSiteDomain }</li>
```

Does that make sense at all? Would this kind of approach require using another function to render each `<li>` separately?

**Comment by danhauk** [comment]
That said, another option would be to just remove the checkmark and make the items not in progress a lighter color:

image

image

I also changed the tense of the verb depending on state here. `Backing up` => `Backed up`

**Comment by jonathansadowski** [comment]
> I agree about the green checkmark on an in progress task is unclear. I like @jonathansadowski suggestion about using the spinner to indicate the task in progress.
> 
> As far as implementation goes, instead of using classes can we switch out the component based on state [here](https://github.com/Automattic/wp-calypso/pull/38053/files#diff-5a90bc5b94ad276cd1f1d73804034db2R233)?
> 
> * if it's in progress, render `<Spinner/>`
> * if complete, render `<Gridicon icon="checkmark"/>`
> 
> ```
> let backupIcon = <Spinner />;
> let restoreIcon;
> 
> if ( 'restoring' === this.state.migrationStatus ) {
>     backupIcon = <Gridicon icon="checkmark-circle" />;
>     restoreIcon = <Spinner />;
> }
> 
> ...
> 
> <li className={ backingUpClass }>{ backupIcon } Backing up from { sourceSiteDomain }</li>
> <li className={ restoringClass }>{ restoreIcon } Restoring to { targetSiteDomain }</li>
> ```
> 
> Does that make sense at all? Would this kind of approach require using another function to render each `<li>` separately?

That's exactly the change I was going to work on in a separate PR :)

**Comment by jonathansadowski** [comment]
@andfinally I took a stab at this in #38180. If you like the approach, feel free to merge that PR into this one.

**Comment by andfinally** [comment]
Thanks a lot Jon and Dan – I was blinded to how I could do the icon by the way I had done it! I've merged Jon's changes in here. As we now have the spinner and the wording change to indicate state the "Wrapping up" item seems superfluous, so I've removed it, leaving us with this much better view:

image

**Pull request merged**